lib/mp4/mp4-generator.js

```diff
@@ -445,23 +465,36 @@ stbl = function(track) {
       0x00, 0x00, 0x00, // compressorname
       0x00, 0x18, // depth = 24
       0x11, 0x11 // pre_defined = -1
-    ]), box(types.avcC, new Uint8Array([
-      0x01, // configurationVersion
-      track.profileIdc, // AVCProfileIndication
-      track.profileCompatibility, // profile_compatibility
-      track.levelIdc, // AVCLevelIndication
-      0xff // lengthSizeMinusOne, hard-coded to 4 bytes
-    ].concat([
-      sps.length // numOfSequenceParameterSets
-    ]).concat(sequenceParameterSets).concat([
-      pps.length // numOfPictureParameterSets
-    ]).concat(pictureParameterSets))), // "PPS"
-            box(types.btrt, new Uint8Array([
-              0x00, 0x1c, 0x9c, 0x80, // bufferSizeDB
-              0x00, 0x2d, 0xc6, 0xc0, // maxBitrate
-              0x00, 0x2d, 0xc6, 0xc0
-            ])) // avgBitrate
-              );
+    ]);
+
+    if (track.sarRatio) {
+      var
+        hSpacing = track.sarRatio[0],
+        vSpacing = track.sarRatio[1],
+        paspBox = box(types.pasp, new Uint8Array([
+          (hSpacing & 0xFF000000) >> 24,
+          (hSpacing & 0xFF0000) >> 16,
+          (hSpacing & 0xFF00) >> 8,
+          hSpacing & 0xFF,
+          (vSpacing & 0xFF000000) >> 24,
+          (vSpacing & 0xFF0000) >> 16,
+          (vSpacing & 0xFF00) >> 8,
+          vSpacing & 0xFF
+        ]));
+
+      result = box(types.avc1, avc1Data,
+        avcCBox,
+        btrtBox,
+        paspBox
```

Comment:
Might be cleaner to push the boxes onto an array and then do a `box.apply(null, boxes)` instead of the if/else.